src/ui/ToggleEditor.tsx

```diff
@@ -429,18 +399,18 @@ class ToggleEditor extends Component<ToggleEditorProps, ToggleEditorState> {
   }
 
   renderBlocks() {
-    let code = this.hasMounted ? this.currentCode : this.props.initialCode;
     return (
       <UpgradedBlockEditor
         cmOptions={this.cmOptions}
-        parse={this.parse}
-        value={code}
+        parse={this.props.language.parse}
+        value={this.state.code}
         onMount={this.handleEditorMounted}
         api={this.props.api}
+        passedAST={this.ast}
+
         appElement={this.props.appElement}
-        language={this.language.id}
+        languageId={this.props.language.id}
         options={this.options}
```

Comment:
`this.options` will only ever contain the set of options that were passed when the component was constructed. But if the component gets updated/re-rendered with a new set of props, `UpgradedBlockEditor` will be using the old set of options. Same problem with `cmOptions`

Reply:
Ditto